src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,399 @@
+    """Print device info."""
+    formatter = OutputFormatter(format, _DEVICE_INFO_KEYS_MAP)
+    device_info = _get_device_info(state.sdk, device_id)
+    if device_info:
+        formatter.echo_formatted_list([device_info])
+
+
+def _get_device_info(sdk, device_id):
+    device = sdk.devices.get_by_id(device_id, include_backup_usage=True).data
+    device["archiveBytes"] = (
+        max(
+            [
+                backupDestination["archiveBytes"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else 0
+    )
+    device["lastBackup"] = (
+        max(
+            [
+                backupDestination["lastBackup"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else None
+    )
+    device["lastCompletedBackup"] = (
+        max(
+            [
+                backupDestination["lastCompletedBackup"]
+                for backupDestination in device["backupUsage"]
+            ]
+        )
+        if len(device["backupUsage"]) > 0
+        else None
+    )
+    return device
+
+
+@devices.group(cls=OrderedGroup)
+@sdk_options(hidden=True)
+def bulk(state):
+    """Tools for managing devices in bulk"""
+    pass
+
+
+@bulk.command(name="list", help="Get information about many devices")
+@click.option(
+    "--active",
+    required=False,
+    type=bool,
+    is_flag=True,
+    default=None,
+    help="Include to return only active devices.",
+)
+@click.option(
+    "--days-since-last-connected",
+    required=False,
+    type=int,
+    help="Return only devices that have not connected in the number of days specified.",
+)
+@click.option(
+    "--org-uid",
+    required=False,
+    type=str,
+    default=None,
+    help="Optionally provide to limit devices to only the ones in the org you specify. Note that child orgs will be included.",
+)
+@click.option(
+    "--drop-most-recent",
+    required=False,
+    type=int,
+    help="Will drop the X most recently connected devices for each user from the result list where X is the number you provide as this argument. Can be used to avoid passing the most recently connected device for a user to the deactivate command",
+)
+@click.option(
+    "--include-backup-usage",
+    required=False,
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Include to return backup usage information for each device (may significantly lengthen the size of the return)",
+)
+@click.option(
+    "--include-usernames",
+    required=False,
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Include to add the username associated with a device to the output",
+)
+@click.option(
+    "--include-settings",
+    required=False,
+    type=bool,
+    default=False,
+    is_flag=True,
+    help="Include to include device settings in output",
+)
+@sdk_options()
+def bulk_list(
```

Comment:
Should be this under the `bulk` command group? I realize it uses the bulk processor, but would users know they need to search under bulk to find the list command?

It does not like this like command takes bulk data but rather it is just async, so it should perhaps just be `code42 devices list`

Reply:
Is this because the bulk stats are helpful?  Does this command take a while?  Even if that is the case, it still might make sense to not this not be in the bulk bucket... I am curious what others think as well

Reply:
I put it under the bulk command group because it deals with devices in bulk. My thought was that we'd have an organization of commands that deal with an individual device (show, deactivate, etc), and then a bulk command for operating on many devices at once (bulk list, bulk deactivate, etc.).

I could move the list command out of the bulk group, but I worry that it might be confusing as it's meant to be used in combination with the other bulk command.

Reply:
The use of run_bulk_process is only for adding device settings, which require an API call per device (i.e. they cannot be retrieved from sdk.devices.get_all()).

Reply:
I see where you are coming from. Listing the devices and showing a device are both top-level commands in the context of this application though, like these others:

`code42 legal-hold list`
`code42 departing-employee list`
`code42 profile list`

and 

`code42 legal-hold show`
`code42 profile show`

etc.

And bulk operations specifically refer to bulk processing a data set, like any WRITE command on a CSV.

I also see listing devices as having other use cases outside of being used with the other bulk command in this PR.

Reply:
Makes sense. I'll move `list` out of the `bulk` group.